libazureinit/src/media.rs

```diff
@@ -77,14 +77,17 @@ const CDROM_VALID_FS: &[&str] = &["iso9660", "udf"];
 
 // Get a mounted device with any filesystem for CDROM
 pub fn get_mount_device() -> Result<Vec<String>, Error> {
-    let mut list_devices: Vec<String> = Vec::new();
-
-    while let Some(device) = block_utils::get_mounted_devices()?
+    let devices = block_utils::get_mounted_devices()?;
+    let list_devices: Vec<String> = devices
         .into_iter()
-        .find(|dev| CDROM_VALID_FS.contains(&dev.fs_type.to_str()))
-    {
-        list_devices.push(device.name);
-    }
+        .filter_map(|dev| {
+            if CDROM_VALID_FS.contains(&dev.fs_type.to_str()) {
+                Some(dev.name)
```

Comment:
It turns out that `dev.name` is not the full path to the device, but rather just [the file name](https://docs.rs/block-utils/0.11.1/src/block_utils/lib.rs.html#183). Given how we're using this library, we might be better off using `fstab` crate directly and do something similar to [what block-utils does](https://docs.rs/block-utils/0.11.1/src/block_utils/lib.rs.html#451) except we would use the [FsEntry](https://docs.rs/fstab/latest/fstab/struct.FsEntry.html) directly.

Reply:
In the end, this makes things so much easier! Will adjust tests accordingly, thank you Jeremy!!